src/Server.php

```diff
@@ -137,6 +137,33 @@ public function handleRequest(ConnectionInterface $conn, Request $request)
             if (strtolower(end($transferEncodingHeader)) === 'chunked') {
                 $stream = new ChunkedDecoder($conn);
             }
+
+            if ($request->hasHeader('Content-Length')) {
+                // remove 'Content-Length' header accordig to: https://tools.ietf.org/html/rfc7230#section-3.3.3
+                $headers = $request->getHeaders();
+                unset($headers['Content-Length']);
+
+                $request = new Request(
+                    $request->getMethod(),
+                    $request->getPath(),
+                    $request->getQueryParams(),
+                    $request->getProtocolVersion(),
+                    $headers
+                );
+            }
+        }
+
+        if ($request->hasHeader('Content-Length')) {
+            $string = $request->getHeaderLine('Content-Length');
+
+            $contentLength = (int)$string;
+            if ((string)$contentLength !== (string)$string) {
+                // Content-Length value is not an integer or not a single integer
+                $this->emit('error', array(new \InvalidArgumentException('The value of `Content-Length` is not valid')));
+                return $this->writeError($conn, 400);
+            }
+
+            $stream = new LengthLimitedStream($conn, $contentLength);
         }
```

Comment:
What if a request has neither chunked transfer encoding nor a `Content-Length` header (such as for simple GET requests)? Is this tested?

Should this be part of this PR? (see also #104)